### ISSUE TITLE
Allow use as FIDO authenticator

### ISSUE THREAD

**Issue by bitgamma** [status-im/hardware-wallet#19]
We could implement, as a separate applet, a FIDO U2F authenticator. Ideally, if the formats are compatible, one would be able to use one of the derived keys as identity for authentification.

**Comment by guylouis** [comment]
I agree this would be a very natural extra feature of the card. And quite easy to add. Are there any good open source implementation for this ? 

I wouldn't either put it on top priority of the to-do-list, unless we identify some usages of U2F linked with adoption of ethereum, dapps, and wallet space. What do you think ?

**Comment by bitgamma** [comment]
There are open source implementations, but I want to explore the possibility of not using separate identity keys for FIDO and instead integrate it with our key tree (using the non-wallet subtree). Of course this would mean doing our own implementation. I also do not consider this to be top priority for now.

**Comment by mulles** [comment]
#79

**Comment by rileyg98** [comment]
For something like this, that's kind of outside the scope of this applet. If you're after a FIDO2 compliant spec, you can freely fork my implementation of this (https://github.com/VivoKey/vk-u2f).

**Comment by rkreutz** [comment]
Hey there. Any plans on supporting this? I'd say even just supporting a very basic implementation with no connection with the main applet keys would already be very helpful and might drive adoption of keycard for other things (not only tied to cryptocurrencies).

**Comment by rkreutz-crypto** [comment]
For future reference, I've managed to use this as an U2F authenticator on my Keycard https://github.com/darconeous/u2f-javacard. Just downloading the CAP from the releases and following the script on the README was enough to have it working on my iOS device.

**Comment by rkreutz** [comment]
For future reference, I've managed to use this as an U2F authenticator on my Keycard https://github.com/darconeous/u2f-javacard. Just downloading the CAP from the releases and following the script on the README was enough to have it working on my iOS device.

**Comment by bitgamma** [comment]
we keep the card open so that you can install any applet on it. We might look into preinstalling it in future batches but there is nothing preventing you to install it using the venerable https://github.com/martinpaljak/GlobalPlatformPro. Just make sure you use the correct globalplatform key which is c212e073ff8b4bbfaff4de8ab655221f

**Comment by willianpaixao** [comment]
I'll try to install it, but my question is whether this feature could be brought as a official/supported Keycard feature.

**Comment by lwinch2006** [comment]
Do I understand correctly that on production Keycards that are sold via keycard.tech website ENC, MAC and DEC keys are set to c212e073ff8b4bbfaff4de8ab655221f?

**Comment by bitgamma** [comment]
> Do I understand correctly that on production Keycards that are sold via keycard.tech website ENC, MAC and DEC keys are set to c212e073ff8b4bbfaff4de8ab655221f?

those cards are for end users and have random ENC, MAC, DEK keys so are effectively locked. Our development cards have the keys you mentioned. Note however than you can use any compatible card (for example JCOP4) and install Keycard on it

**Comment by guylouis** [comment]
we will soon sell our development card on keycard.tech to help anyone who want to develop on javacard, but in the meanwhile if you need one you can just get in touch with us 
for instance you can reach us here: get@keycard.tech

**Comment by lwinch2006** [comment]
@guylouis
FYI. There are wrong applets AID in compiled (at least latest) CAP file in Releases and in `build.gradle` file in the repository. (See attached picture)
Because of this keycard applet is not found and card is not recognized as Keycard.

<img width="986" alt="Image" src="https://github.com/user-attachments/assets/f5464dba-fdab-4bae-be90-3b314ccca813" />
<img width="429" alt="Image" src="https://github.com/user-attachments/assets/99970016-9d88-47fd-b107-962f231f051a" />
<img width="491" alt="Image" src="https://github.com/user-attachments/assets/0e4cb610-2f24-4de6-9dbf-420e3463319e" />

**Comment by bitgamma** [comment]
You are confusing applet AID (set at compile time) with instance AID (set at installation time). This is because the same applet can be installed multiple times on the same card and each instance needs its unique AID. When you install the applet you need to specify the instance AID, which for the Keycard and Keycard Cash applets matches the applet AID + '01'

**Comment by lwinch2006** [comment]
@bitgamma 
mmm, I have tried to set any other AID, `including compile AID + 01` during applet installation, but 
* during installation error shown
* applet anyway getting installed but with compile AID not that pointed out in command arguments

But maybe its something that not works in my setup

<img width="661" alt="Image" src="https://github.com/user-attachments/assets/9a8ee8eb-a761-42e7-bdb2-8bc668b130dd" />

**Comment by lwinch2006** [comment]
@bitgamma 
Sorry its my fault,
GlobalPlatform Pro utility has another parameter --create to point instance AID.
So I think with `--applet compile AID` and `--create instance AID` all will work
